### PR TITLE
Add kibana monitoring

### PR DESCRIPTION
**Pull request by erebe**
+ Add kibana monitoring 
+ Some fix with metrics cleanup
+ Remove some dead code
+ Add some comment for future work

### CODE REVIEW

File: cmd/discovery.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ type esnode struct {
 	version string
 }
 
+const ELASTICSEARCH_SERVICE = "elasticsearch-all"
```
Comment (erebe) [inline cmd/discovery.go]: It is criteo specific, so it should be removed or the project not be open-source

Comment (geobeau) [reply cmd/discovery.go]: You can set it in the flags

File: cmd/probing.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func probeNode(node *esnode, updateProbingPeriod time.Duration) error {
 	if err != nil {
 		log.Debug("Probing failed for ", node.name, ": ", probingURL, " ", err.Error())
 		log.Error(err)
-		nodeAvailabilityGauge.WithLabelValues(node.cluster, node.name).Set(0)
+		elasticNodeAvailabilityGauge.WithLabelValues(node.cluster, node.name).Set(0)
```
Comment (geobeau) [inline cmd/probing.go:29]: For another PR: usually it's better to have availability as counter (total - failure)

File: cmd/probing.go [hunk 3]
```diff
@@ -50,8 +50,56 @@ func probeNode(node *esnode, updateProbingPeriod time.Duration) error {
...
+	}
+
+	body, readErr := ioutil.ReadAll(resp.Body)
+	if readErr != nil {
+		kibanaNodeAvailabilityGauge.WithLabelValues(node.cluster, node.name).Set(0)
+		return fmt.Errorf("kibana Probing failed: %s", readErr)
+	}
+
+	var p fastjson.Parser
+	v, jsonErr := p.Parse(string(body))
```
Comment (geobeau) [inline cmd/probing.go]: What does `v` mean? alors maybe fastjson.Parser.Parse(string(body))

File: cmd/probing.go [hunk 4]
```diff
@@ -50,8 +50,56 @@ func probeNode(node *esnode, updateProbingPeriod time.Duration) error {
...
+	if jsonErr != nil {
+		kibanaNodeAvailabilityGauge.WithLabelValues(node.cluster, node.name).Set(0)
+		return fmt.Errorf("kibana Probing failed: %s", jsonErr)
+	}
+	var nodeState = string(v.GetStringBytes("status", "overall", "state"))
```
Comment (geobeau) [inline cmd/probing.go]: `nodeState := `

File: cmd/probing.go (hunk 3 above)
Comment (erebe) [reply cmd/probing.go]: updated v to json. can't use fastjson.Parser.Parse as it need to be a pointer

**Review by geobeau — APPROVED** [review]

**Pull request merged**